Re-evaluate golang runtime

Go has been using modules instead of `dep` since at least 1.14. As a result `dep` has been archived and is no longer maintained.

Fast forward to Go 1.16, and what may have been considered legal is no longer permitted. One of the dependencies brought in by `dep` contains an invalid pointer:
```
pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:103:21: cannot use sv (type *semver.Version) as type semver.Version in field value
pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:122:16: invalid type assertion: c.(semver.Version) (non-interface type *semver.Constraints on left)
pkg/mod/github.com/golang/dep@v0.5.4/gps/constraint.go:149:4: undefined: semver.Constraint
error building image: error building stage: waiting for process to exit: exit status 2
```

I've been able to work around this by forcing the docker build image to use `golang:1.15.8-alpiine` instead of the default `golang:latest` however the correct approach should be to replace `dep` with `go mod`.

--- a/go/runtime.yaml
+++ b/go/runtime.yaml
@@ -46,8 +46,7 @@ spec:
         ENV GIT_SSH_COMMAND "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
         RUN apk --no-cache add git ca-certificates openssh \
         && go get github.com/triggermesh/aws-custom-runtime \
-        && go get github.com/triggermesh/knative-lambda-runtime/go \
-        && go get github.com/golang/dep/...
+        && go get github.com/triggermesh/knative-lambda-runtime/go
         RUN mv /go/bin/go /go/bin/bootstrap
         WORKDIR /go/src/handler
         COPY . .
@@ -55,9 +54,11 @@ spec:
             eval "\$(ssh-agent -s)"; \
             ssh-add $HOME/.ssh/id_$(inputs.params.SSH_KEY); \
         fi \
-        && if [ -f "Gopkg.toml" ]; then dep ensure; fi \
-        && go get -v \
-        && go install
+        && if [ ! -f "go.mod" ]; then \
+             go mod init handler; \
+           fi \
+        && go get -d -v \
+        && go build -o handler
 
         FROM alpine
         WORKDIR /opt
@@ -65,6 +66,7 @@ spec:
         ENV _HANDLER "handler"
         COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
         COPY --from=0 /go/bin/ /opt
+        COPY --from=0 /go/src/handler/handler /opt
         ENTRYPOINT ["/opt/aws-custom-runtime"]
       EOF
   - name: export

